### PR TITLE
Speed up cmd/protoc.

### PR DESCRIPTION
**Pull request by petermattis**
Work around https://github.com/golang/go/issues/12327 by moving the bulk
of protoc into a library. We still relink protoc on every go install,
but the relink only takes 1 second.

### CODE REVIEW

File: import.sh [hunk 1]
```diff
@@ -162,7 +162,7 @@ google/protobuf/compiler/zip_writer.cc
 
 # symlink so cgo compiles them
 for file in $PROTOC_SOURCES; do
-  ln -sf ../../internal/src/$file cmd/protoc/
+  ln -sf ../internal/src/$file libprotoc
```
Comment (tamird) [inline import.sh:165]: hm, why did this lose one `../`?

Comment (tamird) [inline import.sh:165]: oh, i see. LGTM

File: libprotoc/srcs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+../internal/src/google/protobuf/compiler/code_generator.cc
```
Comment (tamird) [inline libprotoc/srcs]: what is this file for?

Comment (petermattis) [inline libprotoc/srcs]: Oops, that was how I did the movement. Removed.

**Pull request merged**